tests/utils.py

```diff
@@ -34,7 +58,7 @@ def run(
     callbacks.append(history)
     kwargs['callbacks'] = callbacks
 
-    with benchmarks(history):
+    with benchmarks(history, spec.work_dir):
```

Comment:
I can't see a way to avoid passing extra information to the benchmarks context manager - the history does not contain this information.